### PR TITLE
[multistage] Support Geometry queries

### PR DESCRIPTION
**Pull request by xiangfu0**
1. Support st-* functions in multi-stage engine
2. Add integration test

### CODE REVIEW

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java [hunk 1]
```diff
@@ -173,29 +173,37 @@ public enum TransformFunctionType {
   SCALAR("scalar"),
 
   // Geo constructors
-  ST_GEOG_FROM_TEXT("ST_GeogFromText"),
-  ST_GEOM_FROM_TEXT("ST_GeomFromText"),
-  ST_GEOG_FROM_WKB("ST_GeogFromWKB"),
-  ST_GEOM_FROM_WKB("ST_GeomFromWKB"),
-  ST_POINT("ST_Point"),
-  ST_POLYGON("ST_Polygon"),
+  ST_GEOG_FROM_TEXT("ST_GeogFromText", ReturnTypes.explicit(SqlTypeName.VARBINARY), OperandTypes.STRING),
```
Comment (walterddr) [inline pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:176]: can we add some test for the auto function name variance generated? (e.g. `ST_GEOG_FROM_TEXT` and the `ST_GeogFromText`)

File: pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java [hunk 2]
```diff
@@ -173,29 +173,37 @@ public enum TransformFunctionType {
...
-  ST_AS_BINARY("ST_AsBinary"),
-  ST_AS_TEXT("ST_AsText"),
+  ST_AS_BINARY("ST_AsBinary", ReturnTypes.explicit(SqlTypeName.VARBINARY), OperandTypes.BINARY),
+  ST_AS_TEXT("ST_AsText", ReturnTypes.VARCHAR_2000_NULLABLE, OperandTypes.BINARY),
 
   // Geo relationship
-  ST_CONTAINS("ST_Contains"),
-  ST_EQUALS("ST_Equals"),
-  ST_WITHIN("ST_Within"),
+  ST_CONTAINS("ST_Contains", ReturnTypes.INTEGER,
```
Comment (walterddr) [inline pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:196]: shouldn't these return boolean?

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/GeoSpatialClusterIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+    FileUtils.deleteDirectory(_tempDir);
+  }
+
+  @Test(dataProvider = "useBothQueryEngines")
+  public void testGetHexagonAddress(boolean useMultiStageQueryEngine)
+      throws Exception {
+    setUseMultiStageQueryEngine(useMultiStageQueryEngine);
+
+    String query = "Select geoToH3(20,102,5) from " + DEFAULT_TABLE_NAME;
```
Comment (walterddr) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/GeoSpatialClusterIntegrationTest.java:145]: let's add unit-tests in pinot-query-runtime

File: pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java [hunk 4]
```diff
@@ -173,29 +173,37 @@ public enum TransformFunctionType {
...
+      OperandTypes.family(ImmutableList.of(SqlTypeFamily.NUMERIC, SqlTypeFamily.NUMERIC, SqlTypeFamily.NUMERIC),
+          ordinal -> ordinal > 1 && ordinal < 4)),
+  ST_POLYGON("ST_Polygon", ReturnTypes.explicit(SqlTypeName.VARBINARY), OperandTypes.STRING),
 
   // Geo measurements
-  ST_AREA("ST_Area"),
-  ST_DISTANCE("ST_Distance"),
-  ST_GEOMETRY_TYPE("ST_GeometryType"),
+  ST_AREA("ST_Area", ReturnTypes.DOUBLE_NULLABLE, OperandTypes.BINARY),
+  ST_DISTANCE("ST_Distance", ReturnTypes.DOUBLE_NULLABLE,
```
Comment (walterddr) [inline pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:187]: Looks like `ST_DISTANCE` `ST_GEOMETRY_TYPE` `ST_POLYGON` `ST_AS_BINARY` `ST_AS_TEXT`, ST_CONTAINS` are not tested.

**Pull request merged**